code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessageAssetDownloader.java

```diff
@@ -41,139 +46,171 @@ class MessageAssetDownloader {
     private final Networking networkService;
     private final DeviceInforming deviceInfoService;
     private final CacheService cacheService;
-    private final String messageId;
     private File assetDir;
     private final String assetCacheLocation;
 
     /**
      * Constructor.
      *
-     * @param assets          {@code ArrayList<String>} of assets to download and cache
-     * @param parentMessageId {@link String} containing the message Id of the requesting message used as a cache subdirectory
+     * @param assets {@code ArrayList<String>} of assets to download and cache
      */
-    MessageAssetDownloader(final List<String> assets, final String parentMessageId) {
+    MessageAssetDownloader(final List<String> assets) {
         this.assetsCollection = assets;
         this.networkService = ServiceProvider.getInstance().getNetworkService();
```

Comment:
`ServiceProvider` allows overriding NetworkService and DeviceInfoService. If any of these services are overridden later, holding references to them  will cause an incorrect service version to be used. Recommend using `SeviceProvider.getInstance().getNetworkService()` and `SeviceProvider.getInstance().getDeviceInfoService()` when required directly.

Reply:
makes sense, i will update this in a new pr